packages/core/core-flows/src/payment/workflows/process-payment.ts

```diff
@@ -15,6 +16,37 @@ export const processPaymentWorkflow = createWorkflow(
       entity: "payment",
       fields: ["id"],
       filters: { payment_session_id: input.data?.session_id },
+    }).config({
+      name: "payment-query",
+    })
+
+    const paymentSessionResult = useQueryStep({
+      entity: "payment_session",
+      fields: ["payment_collection_id"],
+      filters: { id: input.data?.session_id },
+    }).config({
+      name: "payment-session-query",
+    })
+
+    const cartPaymentCollection = useQueryStep({
+      entity: "cart_payment_collection",
+      fields: ["cart_id"],
+      filters: {
+        payment_collection_id:
+          paymentSessionResult.data[0].payment_collection_id,
+      },
+    }).config({
+      name: "cart-payment-query",
+    })
+
+    when({ cartPaymentCollection }, ({ cartPaymentCollection }) => {
+      return !!cartPaymentCollection.data.length
+    }).then(() => {
+      completeCartWorkflow.runAsStep({
```

Comment:
The reason this logic was separated into two workflows was that the payment processing should not be affected by any subsequent actions we might want to perform, here cart completion.

What happens if the cart completion fails now that is is run as a nested workflow?

Reply:
Nice, that makes sense!